### PR TITLE
Remove customergroup blocked articles in basket

### PR DESCRIPTION
**Pull request by shyim**
| Questions | Answers |
| --- | --- |
| BC breaks? | no |
| Tests pass? | yes |
| Related tickets? | There was a issue. i cant find it lel |
| How to test? | Add article to cart which is blocked for merchants, login to a merchant account in checkout and the article should be removed |
## Description

Please describe your pull request:

We had some troubles with customers, because some products are blocked for merchants and merchants has ordered the products with that bug :)

### CODE REVIEW

**Comment by janbuecker** [comment]
We should prevent adding it too. So please consider this case in `sAddArticle()`.

**Comment by shyim** [comment]
@janbuecker added

**Comment by dnoegel** [comment]
This will currently iterate all items in the basket and perform those queries. This should be changed in a way, that only one query is needed to determine all items, which need to be removed. The deletion of items then could happen in a transaction, for example.

**Comment by shyim** [comment]
@dnoegel updated :)

File: engine/Shopware/Core/sBasket.php [hunk 1]
```diff
@@ -2774,4 +2780,27 @@ private function getBasketQuantity($quantity, $basketProduct, $article)
...
         return $newQuantity;
     }
+
+    /**
+     * @param $articleId
+     * @return string
+     */
+    private function isProductBlockedByCustomerGroup($articleId)
+    {
+        return $this->db->fetchOne('SELECT 1 FROM s_articles_avoid_customergroups WHERE customergroupId = ? AND articleID = ?', [
```
Comment (mitelg) [inline engine/Shopware/Core/sBasket.php]: I think it is more readable, if you use something like :customerGroupId instead of ?

**Review by mitelg — CHANGES_REQUESTED** [review]

File: engine/Shopware/Core/sBasket.php (hunk 1 above)
Comment (shyim) [reply engine/Shopware/Core/sBasket.php]: Aye Aye, Captain!

**Review by mitelg — APPROVED** [review]

**Comment by janbuecker** [comment]
We've changed the code entirely so that it resulted in a new commit 474703992c4e2b91e5a099798737b6b5a4804bb4.

**Pull request closed without merging**